cmd/juju/cloud/add.go

```diff
@@ -74,20 +74,23 @@ type AddCloudCommand struct {
 	// default it just calls the correct provider's Ping method.
 	Ping func(p environs.EnvironProvider, endpoint string) error
 
+	// CloudCallCtx contains context to be used for any cloud calls.
+	CloudCallCtx       *context.CloudCallContext
 	cloudMetadataStore CloudMetadataStore
 }
 
 // NewAddCloudCommand returns a command to add cloud information.
 func NewAddCloudCommand(cloudMetadataStore CloudMetadataStore) *AddCloudCommand {
-	cloudCallCtx := context.NewCloudCallContext()
+	cmd := AddCloudCommand{
+		cloudMetadataStore: cloudMetadataStore,
+		CloudCallCtx:       context.NewCloudCallContext(),
+	}
 	// Ping is provider.Ping except in tests where we don't actually want to
 	// require a valid cloud.
-	return &AddCloudCommand{
-		cloudMetadataStore: cloudMetadataStore,
-		Ping: func(p environs.EnvironProvider, endpoint string) error {
-			return p.Ping(cloudCallCtx, endpoint)
-		},
+	cmd.Ping = func(p environs.EnvironProvider, endpoint string) error {
```

Comment:
This can continue to be in-lined, right?

cloudCallCtx := context.NewCloudCallContext()
return &AddCloudCommand{
	cloudMetadataStore: cloudMetadataStore,
	CloudCallCtx: cloudCallCtx,
	Ping: func(p environs.EnvironProvider, endpoint string) error {	
		return p.Ping(cloudCallCtx, endpoint)	
	},
}